src/cliframe/components/TriCliFrame.tsx

```diff
@@ -0,0 +1,190 @@
+    //         case "Enter":
+    //             this.process()
+    //             this.hide_and_clear()
+    //             break
+    // 
+    //         case "j":
+    //             /* Just like hitting enter, but we need to keep firefox
+    //              * from focusing the omnibar. */
+    //             if (keyevent.ctrlKey) {
+    //                 keyevent.preventDefault()
+    //                 keyevent.stopPropagation()
+    //                 this.process()
+    //                 this.hide_and_clear()
+    //             }
+    //             break
+    // 
+    //         case "m":
+    //             /* Just like hitting enter, but we need to keep firefox
+    //              * from doing whatever it does with the key. */
+    //             if (keyevent.ctrlKey) {
+    //                 keyevent.preventDefault()
+    //                 keyevent.stopPropagation()
+    //                 this.process()
+    //                 this.hide_and_clear()
+    //             }
+    //             break
+    // 
+    //         case "a":
+    //             if (keyevent.ctrlKey) {
+    //                 keyevent.preventDefault()
+    //                 keyevent.stopPropagation()
+    //                 this.setCursor()
+    //             }
+    //             break
+    // 
+    //         case "e":
+    //             if (keyevent.ctrlKey) {
+    //                 keyevent.preventDefault()
+    //                 keyevent.stopPropagation()
+    //                 this.setCursor(this.state.commandlineContents.length)
+    //             }
+    //             break
+    // 
+    //         case "Tab":
+    //             /* Keep the input from losing focus*/
+    //             keyevent.preventDefault()
+    //             keyevent.stopPropagation()
+    //             if (keyevent.shiftKey) {
+    //                 this.activeCompletions.forEach(comp => comp.prev())
+    //             } else {
+    //                 this.activeCompletions.forEach(comp => comp.next())
+    //             }
+    //             break
+    // 
+    //         case " ":
+    //             const command = this.getCompletion()
+    //             this.activeCompletions.forEach(comp => (comp.completion = undefined))
+    //             if (command) this.fillcmdline(command, false)
+    //             break
+    // 
+    //         case "Escape":
+    //             keyevent.preventDefault()
+    //             this.hide_and_clear()
+    //             break
+    //     }
+    // }
+    // 
+    // private setCursor(n = 0) {
+    //     this.inputRef.current.setSelectionRange(n, n, "none")
+    // }
+    // 
+    // private hide_and_clear() {
+    //     // Delete all completion sources - I don't think this is required, but this
+    //     // way if there is a transient bug in completions it shouldn't persist.
+    //     this.activeCompletions = []
+    // 
+    //     this.setState({
+    //         commandlineContents: "",
```

Comment:
ah, I think all of this commented out stuff might be a dead heron - it looks like saulrh was in the middle of moving towards using Redux.